### DISCUSSION TITLE
How to style select options using styled components?

### DISCUSSION THREAD

**Discussion by branislav-brincko** [mantinedev/mantine#6636] [Q&A]
We use styled components with Mantine in our project. Overriding styles usually works great as shown [here](https://help.mantine.dev/q/third-party-styles). The problem appears for "portal components", for example select options. The following does not work, because options are rendered in portal and thus styled components classes/styles are not applied.

![image](https://github.com/user-attachments/assets/6b2b8090-48bc-4c76-981d-3c4c9d19fcd4)

This limitation is mentioned at the bottom of [this](https://mantine.dev/styles/css-modules/) page. I am fine with using styled components `createGlobalStyle` but I am not able to target specific size option - if I do the following the font-size gets applied to all size variants of the select component - I want to apply this only to "lg" and "xl" variants. Is there any workaround for this? Thanks in advance 🙇 

![image](https://github.com/user-attachments/assets/c1d04bce-9d82-415c-8b3d-ec924cadbeef)

**Comment by rtivital** [comment]
In this case, the only option is to set `withinPortal={false}` on Select component.

**Reply by branislav-brincko** [reply]
Thank you very much. 
Are there any drawbacks of this solution? Maybe performance (if there were many elements, for example) 🤔 ?

**Reply by rtivital** [reply]
You can learn about portal features here – https://mantine.dev/core/portal/
